test/unit/modules/compiler/codegen.spec.js

```diff
@@ -74,6 +74,41 @@ describe('codegen', () => {
     )
   })
 
+  it('generate v-local directive', () => {
+    assertCodegen(
+      '<div v-local:value="some.deep.prop">{{value}}</div>',
+      `with(this){return ((function(value){return _c('div',{},[_v(_s(value))])})(some.deep.prop))}`
+    )
+  })
+
+  it('generate multi v-local directive', () => {
+    assertCodegen(
+      '<div v-local:foo="some.deep.prop" v-local:bar="other.deep.prop">{{foo}}{{bar}}</div>',
+      `with(this){return ((function(foo,bar){return _c('div',{},[_v(_s(foo)+_s(bar))])})(some.deep.prop,other.deep.prop))}`
+    )
+  })
+
+  it('generate v-local directive with v-if', () => {
+    assertCodegen(
+      '<div v-local:foo="some.deep.prop" v-if="foo.show">{{foo.value}}</div>',
+      `with(this){return ((function(foo){return (foo.show)?_c('div',{},[_v(_s(foo.value))]):_e()})(some.deep.prop))}`
+    )
+  })
+
+  it('generate v-local directive with v-for', () => {
```

Comment:
I think we need another test for combining v-for, scoped slot and v-local together as scoped vFor is handled separately IIRC.

Reply:
updated,
1. add some test for scoped slot 
2. fix invalid v-local in scoped-slot with template tag

Reply:
After some deeper thoughts, `v-local` might be harder than I originally estimated. The root problem is it adds a new way to introduce new variable in template. After this, we will have three way to introduce new variables:

1. v-for
2. slot-scope
3. v-local

With 1 and 2 we already have some ordering issue, https://github.com/vuejs/vue/issues/6817. I would say more variable introduction isn't scalable in current implementation. For example, `<div v-local:items="foo.deep.prop" v-for="item in items">` is still missing in test. And supporting such usage will require yet another patch in `genFor`.